thriftbp/server.go

```diff
@@ -57,6 +63,16 @@ type ServerConfig struct {
 	// Deprecated: No-op for now, will be removed in a future release.
 	Timeout time.Duration
 
+	// Optional, This duration is used to set both the read and write idle timeouts
+	// for the thrift.TServerSocket used by the baseplate server.
+	//
+	// This is an experimental configuration and is subject to change or deprecation
+	// without notice. Setting a socket timeout will also override the default thrift
+	// server logger to one that emits metrics instead of logs in the event of a
+	// socket disconnect. A zero value means I/O read or write operations
+	// will not time out.
+	ThriftSocketTimeout time.Duration
```

Comment:
this is already inside `thriftbp.ServerConfig` so `Thrift-` prefix in the name is kinda redundant.

I would also suggest to just keep using `Timeout` above, and keep the "Deprecated:" comment. so it's still deprecated (if people try to use it, staticcheck will complain, and they can add a `//go:lint` line to suppress the complaint when they know they do need it). we can remove the deprecated comment if we decide that we want to keep it long term.

we also need to add some clarifications to how this is used in `NewBaseplaseServer` vs. `NewServer`:

1. The timeout itself is used by both.
2. The logger behavior currently mentioned here only applies to `NewBaseplateServer`.

Reply:
Based on the discussion I think we're intentionally using a new configuration field so that existing usages of `Timeout` can continue to function as a noop. Re-using the field would undo that.

Reply:
That's fair, but my points about marking it as deprecated and the `Thrift-` prefix in the name is redundant still stand.